cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -424,12 +417,19 @@ impl PathEnvironment {
             RawManifest::new_documented(all_systems.as_slice(), customization)
         };
 
-        let mut environment = Self::write_new_unchecked(
+        let core_env_tempdir = tempfile::tempdir_in(&flox.temp_dir).unwrap();
+        let core_environment = CoreEnvironment::from_contents_unchecked(
+            CanonicalPath::new(core_env_tempdir.path())
+                .expect("should find temptdir created in the same function"),
+            &manifest.to_string(),
+            None,
+        )?;
+
+        let mut environment = Self::init_from_parts_unchecked(
             flox,
-            pointer,
             dot_flox_parent_path,
-            temp_dir,
-            manifest.to_string(),
+            pointer,
+            &core_environment,
```

Comment:
suggestion blocking: I didn't look into exactly why it's needed with you refactored `write_new_unchecked`, but this is more complex and more inefficient, so I'd be inclined to revert it